nixos/modules/virtualisation/glance.nix

```diff
@@ -0,0 +1,199 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.virtualisation.glance;
+  commonConf = ''
+    [database]
+    connection = mysql://glance:${cfg.dbPassword}@${cfg.dbHost}/glance
+    notification_driver = noop
+
+    [keystone_authtoken]
+    auth_uri = http://localhost:5000
+    auth_url = http://localhost:35357
+    auth_plugin = password
+    project_name = service
+    project_domain_id = default
+    user_domain_id = default
+    username = ${cfg.serviceUsername}
+    password = ${cfg.servicePassword}
+
+    [glance_store]
+    default_store = file
+    filesystem_store_datadir = /var/lib/glance/images/
+  '';
+  glanceApiConf = pkgs.writeText "glance-api.conf" ''
+    ${commonConf}
+
+    [paste_deploy]
+    flavor = keystone
+    config_file = ${cfg.package}/etc/glance-api-paste.ini
+  '';
+  glanceRegistryConf = pkgs.writeText "glance-registry.conf" ''
+    ${commonConf}
+
+    [paste_deploy]
+    config_file = ${cfg.package}/etc/glance-registry-paste.ini
+  '';
+in {
+
+  options.virtualisation.glance = {
+    package = mkOption {
+      type = types.package;
+      example = literalExample "pkgs.glance";
+      description = ''
+        Glance package to use.
+      '';
+    };
+
+    enableSingleNode = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        This option enables Glance as a single-machine installation.
+        That is, all of Glance's components are enabled on this
+        machine. This is useful for evaluating and experimenting with
+        Glance.
```

Comment:
In fact, we are currently only support single node. I think we could address multinode setup in future PR.
I update the description to be more explicit about this limitation